src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../TodosContextProvider/TodosContextProvider';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const { id, title, completed } = todo;
+  const [isBeingEdited, setIsBeingEdited] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const handleChangeOfTitle = ((
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setNewTitle(event.target.value);
+  });
+
+  const handleTodoCompleting = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const todosCopy = [...todos];
+    const searchedTodo = todos.find(({ id: todoId }) => todoId === id) as Todo;
+
+    searchedTodo.completed = event.target.checked;
+
+    setTodos(todosCopy);
+  };
+
+  const handleTodoDeleting = () => {
+    const todosCopy = [...todos].filter(({ id: todoId }) => todoId !== id);
```

Comment:
The `filter` array method returns a copy of the array, so you don't have to do it manually.